### PR TITLE
Changelog for 0.10

### PR DESCRIPTION
**Pull request by aquamatthias**
@philipnrmn please look into this changelog as well - wording is sometime hard...

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -19,9 +40,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+
+#### Enhance the reported metrics 
+
+The metrics endpoint (`/metrics`) gives valuable insight into the system behavior.
+We now have meters for all service objects as well as the number applications and groups in the system.
+
+#### Decouple API (un)marshaling types from internal models
+
+With this change we can control API changes and are free to change internal models.
+This will cahnge nothing in the current API, but will help a lot while we create the v3 API.
```
Comment (gkleiman) [inline changelog.md]: s/cahnge/change/

File: changelog.md [hunk 2]
```diff
@@ -1,15 +1,36 @@
...
+#### The FrameworkId is invalidated, if the scheduler triggers an error
+
+If the scheduler reports an error, the framework id is removed from the backing persistence store.
+This is important because on certain kinds of framework errors (such as exceeding the framework failover timeout),
+the scheduler may never re-register with the saved FrameworkID until the leading Mesos master process is killed.
+
+#### Handle requests with missing Accept and Content-Type headers correctly 
+
+If an http request is made without setting appropriate request headers, the Accept and Content-Type header 
+is set automatically to `application/json`
```
Comment (gkleiman) [inline changelog.md]: s/http/HTTP/
Missing period at the end of the line.

File: changelog.md [hunk 3]
```diff
@@ -1,15 +1,36 @@
...
 
-##### Restrict the number of concurrent HTTP requests
+#### Administratively zeroing the task launch delay
+
+It is now possible to reset the task launch delay via the REST endpoint `/v2/queue/{app_id}/delay`.
+The UI displays two additional possible statuses: "Delayed" and "Waiting". 
+The "Delayed" status also displays a tooltip showing the remaining time until the next launch attempt.
+The App page now allows the user to reset the task launch delay for a "Delayed" app, thus forcing a new immediate launch attempt.
+
+#### The FrameworkId is invalidated, if the scheduler triggers an error
```
Comment (gkleiman) [inline changelog.md]: does the scheduler trigger or report an error?

File: changelog.md [hunk 4]
```diff
@@ -19,9 +40,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+This test suite runs with every commit on every pull request.
+This test suite is our safety net, to ensure compatibility while introducing new functionality in future versions.
+
+#### Code coverage reporting using coveralls
+
+See our github page https://github.com/mesosphere/marathon or go directly to  
+https://coveralls.io/github/mesosphere/marathon to see our code coverage reporting.
+With every Pull Request we try to make sure, this coverage will be increased.  
+
+#### Improved the release process of marathon itself
```
Comment (gkleiman) [inline changelog.md]: s/marathon/Marathon/

File: changelog.md [hunk 5]
```diff
@@ -1,15 +1,36 @@
...
 
-Please look at the following changes to check whether you have to verify your setup before upgrading:
+We tested this release against Mesos version 0.22.1. Thus, this is the recommended
+Mesos version for this release.
 
-* If an http request is made without setting appropriate request headers, the Accept and Content-Type header 
-  is set automatically to `application/json`
+### Notes 
+
+This will be the last release that supports Java 6, since Java 6 and Java 7 have reached end of Life.
```
Comment (philipnrmn) [inline changelog.md]: s/Life/life/

File: changelog.md (hunk 3 above)
Comment (philipnrmn) [inline changelog.md]: Lose the comma in this heading

File: changelog.md [hunk 6]
```diff
@@ -19,9 +40,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
 which means that this operation can be retried.
 You can turn on this feature by setting the parameter `--http_max_concurrent_requests`.
 
+#### Serialize all concurrent change requests
+ 
+In former versions of Marathon it was possible, that multiple concurrent changes of the same application definition
```
Comment (philipnrmn) [inline changelog.md]: Lose the comma

File: changelog.md [hunk 7]
```diff
@@ -19,9 +40,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+will lead to an inconsistent state of that application. All changes of groups or application definitions now
```
Comment (philipnrmn) [inline changelog.md]: s/^will/would/

File: changelog.md [hunk 8]
```diff
@@ -19,9 +40,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+get serialized, so the resulting state is always consistent.
+
+#### Enhance the reported metrics 
+
+The metrics endpoint (`/metrics`) gives valuable insight into the system behavior.
+We now have meters for all service objects as well as the number applications and groups in the system.
+
+#### Decouple API (un)marshaling types from internal models
```
Comment (philipnrmn) [inline changelog.md]: I prefer 'unmarshalling' to 'unmarshaling' (note both are correct)

File: changelog.md [hunk 9]
```diff
@@ -19,9 +40,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+
+With this change we can control API changes and are free to change internal models.
+This will cahnge nothing in the current API, but will help a lot while we create the v3 API. 
+
+#### Added integration tests for all API endpoints
+
+We now have an integration test suite, that covers all endpoints with almost all possible parameter combinations.
```
Comment (philipnrmn) [inline changelog.md]: Lose the comma

File: changelog.md [hunk 10]
```diff
@@ -19,9 +40,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+This test suite runs with every commit on every pull request.
+This test suite is our safety net, to ensure compatibility while introducing new functionality in future versions.
```
Comment (philipnrmn) [inline changelog.md]: Lose the comma

File: changelog.md [hunk 11]
```diff
@@ -1,15 +1,42 @@
...
+This version picks instances to kill based on defined constraints and will not violate those. 
+
+#### Handle requests with missing Accept and Content-Type headers correctly 
+
+If an HTTP request is made without setting appropriate request headers, the Accept and Content-Type header 
+is set automatically to `application/json`.
+
+#### Option to restrict the number of concurrent HTTP requests
 
 With this version of Marathon it is possible to restrict the number of concurrent http requests,
```
Comment (kolloch) [inline changelog.md]: s/http/HTTP/

**Comment by kolloch** [comment]
This is not online a changelog but also some kind of newsletter regarding Marathon because it contains references to improvements which do not matter in production (better test infrastructure etc.)

Maybe we should separate these things into separate sections "User-visible changes"/"Infrastructure" or similar.

**Comment by aquamatthias** [comment]
@kolloch I am open to suggestions...

**Comment by kolloch** [comment]
@aquamatthias I know! It's a pleasure to work with you :)

File: changelog.md [hunk 12]
```diff
@@ -19,9 +46,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+- #1709 - Concurrent changes to the same AppDefinition will lead to inconsistent state
+- #1669 - Inconsistent content-type and response from REST API 
+- #1660 - The GUI doesn't allow creating apps with 0 instances, but it is possible through the v2 API 
 - #1647 - Keep the rest-api return format consistent when request headers without "Accept: application/json"
+- #1397 - Kill tasks with scaling does not update group
+- #1654 - Marathon 0.8.1 - API - /v2/tasks 
+- #555  - Constraints not satisfied when scaling down
+- #650  - deployment_success/failed should contain deployment info  
+- #1316 - Invalidate framework ID if registration triggers an error  
+
```
Comment (gkleiman) [inline changelog.md]: After #1807 gets merged, add:
- #1254 - Add flag to define Marathon PORT prefix

File: changelog.md [hunk 13]
```diff
@@ -19,9 +46,56 @@ If the limit of concurrent requests is reached, a HTTP 503 Service Temporarily U
...
+See our github page https://github.com/mesosphere/marathon or go directly to  
+https://coveralls.io/github/mesosphere/marathon to see our code coverage reporting.
+With every Pull Request we try to make sure, this coverage will be increased.  
+
+#### Improved the release process of Marathon itself
+
+Several changes are made to the sbt build file, but also to the infrastructure that is performing the release.
+We now have a more reliable way of pushing distributions.
+
+
```
Comment (gkleiman) [inline changelog.md]: After #1807 gets merged, please add:

#### Custom prefix for automatically created environment variables
It is now possible to specify a custom prefix via the `--env_vars_prefix`
command line flag.  this prefix will be added to the name of task's environment
variables created automatically by Marathon (e.g., `PORT`, `PORTS`)

_Note: This prefix will not be added to variables that are already prefixed,
such as `MESOS_TASK_ID` and `MARATHON_APP_ID`

File: changelog.md [hunk 14]
```diff
@@ -1,17 +1,53 @@
...
+Constraints in Marathon is a powerful mechanism. 
+While the constraints are maintained when scaling up, previous versions of Marathon did not do so while scaling down.
+This version picks instances to kill based on defined constraints and will not violate those. 
 
-With this version of Marathon it is possible to restrict the number of concurrent http requests,
+#### Custom prefix for automatically created environment variables
+
+It is now possible to specify a custom prefix via the `--env_vars_prefix`
+command line flag.  this prefix will be added to the name of task's environment
```
Comment (gkleiman) [inline changelog.md]: s/.  t/. This/

**Pull request merged**